pkgs/development/tools/parsing/flex/default.nix

```diff
@@ -19,27 +19,30 @@ stdenv.mkDerivation rec {
         + "/tools/flex/patches/200-build-AC_USE_SYSTEM_EXTENSIONS-in-configure.ac.patch";
     sha256 = "1aarhcmz7mfrgh15pkj6f7ikxa2m0mllw1i1vscsf1kw5d05lw6f";
   })];
-  nativeBuildInputs = [ autoreconfHook help2man ];
 
+  nativeBuildInputs = [ buildPackages.stdenv.cc autoreconfHook help2man ];
   buildInputs = [ bison ];
-
-  propagatedBuildInputs = [ m4 ];
+  nativePropagatedBuildInputs = [ m4 ];
 
   postConfigure = stdenv.lib.optionalString (stdenv.isDarwin || stdenv.isCygwin) ''
     sed -i Makefile -e 's/-no-undefined//;'
   '';
 
   crossAttrs = {
```

Comment:
I wan't to get rid of `crossAttrs`, so could you make all the inner attributes conditional "top-level" ones instead?